src/Controller/AbstractCsvImportController.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2015 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\Controller;
+
+use Contao\CoreBundle\Exception\CsvImportErrorException;
+use Contao\CoreBundle\Exception\RedirectResponseException;
+use Contao\CoreBundle\Exception\ResponseException;
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Contao\FileUpload;
+use Doctrine\DBAL\Connection;
+use Contao\CoreBundle\Util\CsvImportUtil;
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * Parent class for the CSV import controllers.
+ *
+ * @author Kamil Kuzminski <https://github.com/qzminski>
+ */
+abstract class AbstractCsvImportController
+{
+    /**
+     * @var Connection
+     */
+    protected $connection;
+
+    /**
+     * @var ContaoFrameworkInterface
+     */
+    protected $framework;
+
+    /**
+     * @var TokenStorageInterface
+     */
+    protected $tokenStorage;
+
+    /**
+     * Constructor.
+     *
+     * @param Connection               $connection
+     * @param ContaoFrameworkInterface $framework
+     * @param TokenStorageInterface    $tokenStorage
+     */
+    public function __construct(
+        Connection $connection,
+        ContaoFrameworkInterface $framework,
+        TokenStorageInterface $tokenStorage
+    ) {
+        $this->connection   = $connection;
+        $this->framework    = $framework;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    /**
+     * Set the database connection
+     *
+     * @param Connection $connection
+     */
+    public function setConnection($connection)
+    {
+        $this->connection = $connection;
```

Comment:
What if I would need a different connection for the child class? We could obviously take out the abstract class from the services list but then every child class would need to define those services again (DRY). They are like the default ones here with an option to override by using setters.